packages/wallet-sdk/src/CoinbaseWalletSDK.ts

```diff
@@ -24,8 +28,12 @@ export class CoinbaseWalletSDK {
   }
 
   public makeWeb3Provider(preference: Preference = { options: 'all' }): ProviderInterface {
-    const params = { metadata: this.metadata, preference };
-    return getCoinbaseInjectedProvider(params) ?? new CoinbaseWalletProvider(params);
+    const window = globalThis as CBWindow;
+    const ethereum = window.ethereum ?? window.top?.ethereum;
```

Comment:
![image](https://github.com/coinbase/coinbase-wallet-sdk/assets/966662/6c198bd6-432e-409a-b63f-56ef146f2ca6)
iirc, there was some issue when accessing `top`.
wonder if there's a better way to address it though instead of try-catch.